### ISSUE TITLE
[Bug] Ripgrep command fails in search and template

### ISSUE THREAD

**Issue by crashmoney** [epwalsh/obsidian.nvim#278]
### 🐛 Describe the bug

First and foremost, thank you for this plugin – it's made working with Markdown files in Neovim much more pleasant :)

I started actively using this plugin a few days ago, and have run into issues with all commands relying upon a finder and ripgrep consistently. Commands like `:ObsidianOpen`, `:ObsidianToday`, `:ObsidianYesterday`, etc. and autocompletion of links work, while `:ObsidianSearch`, `:ObsidianTemplate` and `:ObsidianQuickSwitch` fail. Calling `:ObsidianTemplate <template name>` does work.

The `fzf.vim` pop-up appears as it should, but doesn't show any files; instead it presents this error message:

```
[Command failed: rg --no-config --type=md --files --sortr=modified <vault directory>]
```

I have `ripgrep` installed using homebrew, and have no problem calling the above command in a normal shell. `/opt/homebrew/bin` and `/opt/homebrew/opt/rg/bin` are included in both my shell PATH and neovim PATH. `fzf.vim` works perfectly, including the `:Rg` command.

I've configured my vaults as such:

```
{
  name = "vault",
  path = os.getenv("HOME") .. <vault path>,
  overrides = {
    templates = {
      subdir = "templates",
    }
  }
}
```

The directory provided by the error message is correct, matching both the actual directory path and what I have in my `init.lua`.

I'm not entirely sure how to proceed from here, and any help would very much be appreciated. Thanks in advance 👍🏻

### Versions

```
> nvim --version

NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1700008891

> nvim --headless -c 'lua require("obsidian").info()' -c q

[obsidian.nvim (v2.4.0)] Commit SHA: 430bee736fc48170362f38ba1217596d241abdaa
[plenary.nvim] Commit SHA: 55d9fe89e33efd26f532ef20223e5f9430c8b0c0
[nvim-cmp] Commit SHA: 538e37ba87284942c1d76ed38dd497e54e65b891
[fzf.vim] Commit SHA: 1e054c1d075d87903647db9320116d360eb8b024
[vim-markdown] Commit SHA: ff07322775a66dd19c05344d7f33fee03a8e9f8c
ripgrep 14.0.3
```

**Comment by epwalsh** [comment]
Hey @crashmoney I'm trying to reproduce this but so far no luck. What does the vault path look like? Is it behind a symlink or does it contains spaces or other odd characters?

**Comment by epwalsh** [comment]
In the meantime I'd suggest switching to telescope, at least for obsidian. Telescope has a much better API, leading to a cleaner integration with obsidian. Integrating with fzf.vim on the other hand requires some hackery, so it's probably more brittle 🫤

**Comment by crashmoney** [comment]
> Hey @crashmoney I'm trying to reproduce this but so far no luck. What does the vault path look like? Is it behind a symlink or does it contains spaces or other odd characters?

Hi @epwalsh, thanks so much for the prompt reply. I have a few vault workspaces linked at the moment – one has spaces, another has odd characters (fire emoji `🔥`) and the last is one word. It's stored in iCloud, so the standard path is:

```
~/Library/Mobile Documents/iCloud~md~obsidian/Documents/My Vault Name
```

**Comment by crashmoney** [comment]
> In the meantime I'd suggest switching to telescope, at least for obsidian. Telescope has a much better API, leading to a cleaner integration with obsidian. Integrating with fzf.vim on the other hand requires some hackery, so it's probably more brittle 🫤

This works! All the commands that didn't work with fzf.vim are alright now. Thank you so much 👍🏻

**Comment by epwalsh** [comment]
@crashmoney yay! Glad you got it working. FWIW I did figure out the issue with fzf.vim, and have a fix in #281.
